packages/webdriverio/src/session/context.ts

```diff
@@ -77,6 +80,22 @@ export class ContextManager extends SessionManager {
     }
 
     #onCommand(event: { command: string, body: unknown }) {
+        /**
+         * update frame context if user switches using 'switchToParentFrame'
+         */
+        if (event.command === 'switchToParentFrame') {
+            return this.#browser.browsingContextGetTree({}).then(({ contexts }) => {
+                if (!this.#currentContext) {
```

Comment:
I believe this could be moved higher so that the code up to this point is never run when it doesn't have to be